core/lib/zksync_core/src/eth_watch/mod.rs

```diff
@@ -131,13 +157,15 @@ impl EthWatch {
         }
     }
 
-    pub async fn run(
+    async fn run(
         &mut self,
         pool: ConnectionPool,
         stop_receiver: watch::Receiver<bool>,
+        limit: Option<usize>,
     ) -> anyhow::Result<()> {
         let mut timer = tokio::time::interval(self.poll_interval);
-        loop {
+        let mut processed_events = 0;
+        while limit.map(|l| processed_events < l).unwrap_or(true) {
```

Comment:
Nit: You can simplify this call chain to `.map_or(true, |l| processed_events < l)`.

Reply:
won't I have to call an `.unwrap()` after that anyway?